sqlglot/dialects/tsql.py

```diff
@@ -37,11 +106,44 @@ class Parser(Parser):
             "CHARINDEX": exp.StrPosition.from_arg_list,
         }
 
-        def _parse_convert(self):
+        var_length_datatype = [
```

Comment:
```suggestion
        VAR_LENGTH_DATATYPES = {
```

convert it into a set

Reply:
Makes sense yes. On the capitalization of the objects i was not sure. Sometimes it is lower case, like in the dialect class. Changed it however to keep it in line within its own subclass.